modmesh/gui/naca.py

```diff
@@ -94,16 +93,49 @@ def calc_naca4_points(number, npoint, open_trailing_edge=False,
     return ret
 
 
-def draw_naca4(world, number, npoint, fac, off_x, off_y, **kw):
+def linear_approximation(world, points):
+    for it in range(points.shape[0] - 1):
+        e = world.add_edge
+        (points[it, 0], points[it, 1], 0, points[it + 1, 0], points[it + 1, 1], 0)
+        print(f"{it}: {e}")
+    return world.nedge
+
+
+def cubic_bezier_approximation(world, points):
+    Vector = core.Vector3dFp64
+
+    for it in range(points.shape[0] - 1):
+        p1 = points[it] + (1 / 3) * (points[it + 1] - points[it])
+        p2 = points[it] + (2 / 3) * (points[it + 1] - points[it])
+
+        b = world.add_bezier(
+            [
+                Vector(points[it, 0], points[it, 1], 0),
+                Vector(p1[0], p1[1], 0),
+                Vector(p2[0], p2[1], 0),
+                Vector(points[it + 1, 0], points[it + 1, 1], 0),
+            ]
+        )
+        b.sample(7)
+
+        print("control point 1: ", p1)
+        print("control point 2: ", p2)
+        print("bezier locus: ", b.nlocus)
+    return world.nbezier
+
+
+def draw_naca4(world, number, npoint, fac, off_x, off_y, use_bezier, **kw):
```

Comment:
Give `use_bezier` a default value of `False`.

Reply:
Open discussion: does it make sense to turn this function `draw_naca4` a class?

Reply:
Yes, I think so.  Turning function `draw_naca4` a class would be better to manage parameters or behaviors for an airfoil. From my view, `number` and `npoint` in `draw_naca4` are static information of an airfoil and others are the factors about drawing an airfoil. They can be defined in a class as variables and functions separately.

Reply:
I think doing the function-to-class refactor is OK with this PR or another.